include/odp/api/spec/comp.h

```diff
@@ -0,0 +1,863 @@
+	 * ODP_COMP_ERR_OUT_OF_SPACE, application should keep on calling
+	 * odp_comp_xxx() API with more output buffer unless call returns
+	 * with ODP_COMP_ERR_NONE or other failure code except
+	 *  ODP_COMP_ERR_OUT_OF_SPACE.
+	 */
+	odp_bool_t last;
+
+	/** Input data range to process */
+	odp_packet_data_range_t in_data_range;
+
+	/** Output packet data range.
+	 * Indicates where processed packet will be written
+	 * where length specifies,
+	 * length of available packet buffer at input, and
+	 * length of data written at output
+	 */
+	odp_packet_data_range_t out_data_range;
+} odp_comp_packet_op_param_t;
+
+/**
+ * Query comp capabilities
+ *
+ * Output comp capabilities on success.
+ *
+ * @param[out] capa   Pointer to capability structure for output
+ *
+ * @retval 0 on success
+ * @retval <0 on failure
+ */
+int odp_comp_capability(odp_comp_capability_t *capa);
+
+/**
+ * Query supported compression algorithm capabilities
+ *
+ * Output algorithm capabilities.
+ *
+ * @param	comp	Compression algorithm
+ * @param[out]	capa	Compression algorithm capability
+ *
+ * @retval <0 on failure
+ */
+int odp_comp_alg_capability(odp_comp_alg_t comp,
+			    odp_comp_alg_capability_t capa);
+
+/**
+ * Query supported hash algorithm capabilities
+ *
+ * Outputs all supported configuration options for the algorithm.
+ *
+ * @param	hash     Hash algorithm
+ * @param	capa     Hash algorithm capability
+ *
+ * @retval <0 on failure
+ */
+int odp_comp_hash_alg_capability(odp_comp_hash_alg_t hash,
+				 odp_comp_hash_alg_capability_t capa);
+
+/**
+ * Initialize comp session parameters
+ *
+ * Initialize an odp_comp_session_param_t to its default values for
+ * all fields.
+ *
+ * @param param   Pointer to odp_comp_session_param_t to be initialized
+ */
+void odp_comp_session_param_init(odp_comp_session_param_t *param);
+
+/**
+ * Compression session creation
+ *
+ * Create a comp session according to the session parameters. Use
+ * odp_comp_session_param_init() to initialize parameters into their
+ * default values.
+ *
+ * @param param             Session parameters
+ *
+ * @retval	Created session or ODP_COMP_SESSION_INVALID on error
+ */
+odp_comp_session_t odp_comp_session_create(const odp_comp_session_param_t *
+					   param);
+
+/**
+ * Comp session destroy
+ *
+ * Destroy an unused session. Result is undefined if session is being used
+ * (i.e. asynchronous operation is in progress).
+ *
+ * @param session           Session handle
+ *
+ * @retval 0 on success
+ * @retval <0 on failure
+ */
+int odp_comp_session_destroy(odp_comp_session_t session);
+
+/**
+ * Comp set dictionary
+ *
+ * Should be called when there is no operation in progress i.e.
+ * before initiating processing of first chunk of data and
+ * after processing of last chunk of data is complete.
+ *
+ * @param		session		Session handle
+ * @param[in, out]	dict		Pointer to dictionary.
+ *					Implementation should update
+ *					length of dictionary at output.
```

Comment:
Where dict length is updated ? Why dict pointer is in and out parameter?

Reply:
Still seems unclear here. Need a precise specification for this behavior.

Reply:
It's unclear what actual role the application has here. If the specified compression algorithm makes use of a dictionary, why wouldn't that be internal to the implementation and simply part of the `odp_comp_session_t`, allocated on `odp_comp_session_create()` and freed on `odp_comp_session_destroy()`? This seems to imply that `odp_comp_session_create()` is incomplete until this extra API is called. 

If the dictionary is internal to the session, then perhaps an `odp_comp_session_reset()` API might be useful to reset the session back to its "freshly created" state, but it's not clear why the application should need to concern itself directly with dictionaries. The same would go for any huffman tables, which are implied elsewhere but not defined here.